cli/install/install.go

```diff
@@ -0,0 +1,572 @@
+		return true
+	}
+	return false
+}
+
+// getLatestTarantoolTag returns latest tag from tarantool repository.
+func getLatestTarantoolTag() (string, error) {
+	cmd := exec.Command("git", "ls-remote", "--tags", "--sort="+
+		"v:refname", "https://github.com/tarantool/tarantool.git")
+	readPipe, writePipe, _ := os.Pipe()
+	cmd.Stdout = writePipe
+	cmd.Stderr = os.Stderr
+	cmd.Start()
+	err := cmd.Wait()
+	writePipe.Close()
+	if err != nil {
+		return "", err
+	}
+
+	var buf bytes.Buffer
+	io.Copy(&buf, readPipe)
+	versions := buf.String()
+	versionsArray := strings.Split(versions, "\n")
+	latestTag := versionsArray[len(versionsArray)-2]
+	latestTag = strings.TrimSuffix(latestTag, "^{}")
+	trimpos := strings.LastIndex(latestTag, "/") + 1
+	latestTag = latestTag[trimpos:]
+	return latestTag, nil
+}
+
+// copy copies file from source to destination.
+func copy(src string, dst string) error {
+	// Read all content of src to data.
+	data, err := ioutil.ReadFile(src)
+	if err != nil {
+		return err
+	}
+	// Write data to dst.
+	err = ioutil.WriteFile(dst, data, 0777)
+	return err
+}
+
+func copyDir(src string, dst string) error {
+	cmd := exec.Command("cp", "--recursive", src, dst)
+	cmd.Start()
+	err := cmd.Wait()
+	return err
+}
+
+// checkDependecie checks if programm is installed.
+func checkDependecie(programm string) bool {
+	cmd := exec.Command(programm, "--version")
+	cmd.Start()
+	err := cmd.Wait()
+	return err == nil
+}
+
+// checkPythonDependencie checks if python packet is installed.
+func checkPythonDependencie(packet string) bool {
+	cmd := exec.Command("pip3", "list")
+	readPipe, writePipe, _ := os.Pipe()
+	cmd.Stdout = writePipe
+	cmd.Stderr = os.Stderr
+	cmd.Start()
+	cmd.Wait()
+	writePipe.Close()
+	var buf bytes.Buffer
+	io.Copy(&buf, readPipe)
+	packets := buf.String()
+	return strings.Contains(packets, packet)
+}
+
+func CheckLibDependencie(library string) bool {
+	cmd := exec.Command("bash", "-c", "ls"+" /lib/*/")
+	if runtime.GOOS == "darwin" {
+		cmd = exec.Command("brew", "list")
+	}
+	if strings.Contains(DetectOsName(), "CentOS") {
+		cmd = exec.Command("yum", "list", "--installed")
+	}
+	readPipe, writePipe, _ := os.Pipe()
+	cmd.Stdout = writePipe
+	cmd.Stderr = os.Stderr
+	cmd.Start()
+	cmd.Wait()
+	writePipe.Close()
+	var buf bytes.Buffer
+	io.Copy(&buf, readPipe)
+	packets := buf.String()
+	return strings.Contains(packets, library)
+}
+
+// checkTtDependencies checks if tt dependencies is installed.
+func checkTtDependencies() bool {
+	programms := []string{"mage", "git", "python3"}
+	var miss_count = 0
+	missing_pack := ""
+	for i := 0; i < len(programms); i++ {
+		if !checkDependecie(programms[i]) {
+			miss_count++
+			missing_pack += " " + programms[i]
+		}
+	}
+	osName := DetectOsName()
+	if miss_count != 0 {
+		log.Error("The operation requires some dependencies.")
+		log.Error("Missing packages: " + missing_pack)
+		if osName == "darwin" || strings.Contains(osName, "freebsd") {
```

Comment:
FreeBSD uses `pkg` package manager, not `brew`.

Reply:
Changed